socket.go

```diff
@@ -36,7 +36,9 @@ type socket struct {
 	retry time.Duration
 	sec   Security
 
-	mu    sync.RWMutex
+	mu      sync.RWMutex
```

Comment:
traditionally, we put the mutex above the group of variables it is supposed to protect.

I'd suggest to keep `mu` on top of the `ids`,`conns`... block and move `closeMu` just above `closedConns`.

Reply:
Good point. `closeMu` is also "protecting" `cancel` though.
To prevent any confusion on the purpose of `closeMu`, it could be more idiomatic to do away with the mutex and use a "quit" channel that `connReaper` and `scheduleRmConns` can select on.

Reply:
SGTM